src/common/AndroidLauncher.ts

```diff
@@ -88,16 +88,24 @@ export class AndroidLauncher {
                     const launchActivity = appConfig?.activity ?? '';
 
                     const targetAppArguments: LaunchArgument[] = appConfig?.launch_arguments ?? [];
-                    return AndroidUtils.launchNativeApp(
-                        compName,
-                        projectDir,
+
+                    targetAppArguments.push({ name: PreviewUtils.COMPONENT_NAME_ARG_PREFIX, value: compName });
+                    targetAppArguments.push({ name: PreviewUtils.PROJECT_DIR_ARG_PREFIX, value: projectDir });
+
+                    if (address) {
+                        targetAppArguments.push({ name: PreviewUtils.SERVER_ADDRESS_PREFIX, value: address });
+                    }
+
+                    if (port) {
+                        targetAppArguments.push({ name: PreviewUtils.SERVER_PORT_PREFIX, value: port });
+                    }
+
+                    return AndroidUtils.launchAppInBootedEmulator(
```

Comment:
Cleanup the logic so that `launchApp` would only take care of launching the app with a given set of launch args.

Previously, `AndroidLauncher` & `IOSLauncher` contained some logic around determining values for a subset of launch arguments and `launchApp` contained some logic around determining values for another subset of launch arguments.

Now `AndroidLauncher` & `IOSLauncher` will produce `all` of the launch arguments and `launchApp` just consumes them